JavatrixTest.java

```diff
@@ -598,6 +598,33 @@ public void timesEqualsValid()
         assertArrayEquals("failure - double arrays are not same",
             matrix, actual.getArray());
     }
+    
+    /**
+     * Test timesEquals function, returns same object.
+     */
+    @Test
+    public void timesEqualsSameObj()
+    {
+        m = 2;
+        n = 3;
+        double scalar = 3.3;
+        double[] a = {1, 2, 3};
+        double[] b = {4, 5, 6};
+        matrix = new double[m][n];
+        matrix[0] = a;
+        matrix[1] = b;
+
+        double[] c = {2, 3, 4};
+        double[] d = {9, 5, 1};
+        double[][] matrixB = new double[m][n];
+        matrixB[0] = c;
+        matrixB[1] = d;
+
+        actual = new Javatrix(matrix);
+        Javatrix actualB = new Javatrix(matrixB);
```

Comment:
You can simplify this code by removing the matrixB and actualB lines - since those are only included in the values test to compare that the values are accurate. This test just checks that the method returns the same object, so you don't need the one to check the values against.